### PR TITLE
#166840885 Users comment on articles

### PR DESCRIPTION
**Pull request by Onyimatics**
#### Description
 - This PR implements article comments section

#### Type of Change
- [x] Feature (Implement article comments section)

#### How Has This Been Tested?
- Clone the repo [ErrorSwag](https://github.com/andela/ah-kifaru-frontend)
- Run `npm install` on your terminal
- Start the server with `npm run start:dev`

#### Checklist:
- [x] create article comments section

#### Screenshot
- Desktop view 
Screenshot 2019-08-20 at 8 23 45 AM

- Mobile view
Screenshot 2019-08-20 at 8 24 48 AM

  
#### Pivotal Stories
[#166840885](https://www.pivotaltracker.com/story/show/166840885)

### CODE REVIEW

**Review by timi-codes — APPROVED** [review]

File: src/components/Comments/CreateComments.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  handleCommentSubmit,
+  handleInputFieldChange,
+  content,
+  isAuthenticated,
+  contentError,
+  handleClear
+}) => (
+  <section>
+    <div className="flex flex-col w-full m-auto  mb-10 mt-10">
+      <p className="mb-3 font-bold text-2xl">COMMENTS</p>
```
Comment (segunolalive) [inline src/components/Comments/CreateComments.js]: this should be a header. Judging from the design of the page, decide what heading level it should be starting from `h1` for the top-level header.

Comment (segunolalive) [reply src/components/Comments/CreateComments.js]: Provide an accessible label for the comment form. eg

```html
<form onSubmit={handleCommentSubmit} data-testid="submit-form" arai-label="comment">
```

File: src/components/Comments/CreateComments.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        <div className="flex flex-col md:flex-row w-full ">
+          <div className="w-full md:pr-6 pr-0">
+            <textarea
+              data-testid="content"
+              className="w-full h-full p-3 border-solid border-gray-200 border-2 "
+              placeholder="Enter Comment..."
+              name="content"
+              value={content}
+              onChange={handleInputFieldChange}
+              disabled={!isAuthenticated ? 'disabled' : ''}
```
Comment (segunolalive) [inline src/components/Comments/CreateComments.js]:
```html
<textarea disabled={!isAuthenticated} />
```

File: src/components/Comments/ViewComments.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import dateFormat from 'dateformat';
+import PropTypes from 'prop-types';
+
+export const ViewComments = ({ allComments, AvatarIcon }) => (
+  <div
```
Comment (segunolalive) [inline src/components/Comments/ViewComments.js]: Make this an article tag. It is a stand-alone content

File: src/store/modules/comments/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  POST_COMMENT_SUCCESS,
+  POST_COMMENT_FAILURE
+} from '../actionTypes';
+
+const removeQuotesFromString = str => str.trim().slice(1, str.length - 1);
+
+export const getCommentPending = () => ({
+  type: GET_COMMENTS_PENDING,
+  payload: {
+    status: 'commentPending',
```
Comment (segunolalive) [inline src/store/modules/comments/actions/index.js:18]: Useful nit-pick.

You could have used your ACTION_TYPES as the corresponding status as well. This gives the benefit of syntax highlighting and autocompletion. Also, helps eliminate bugs.

File: src/store/modules/comments/actions/index.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+export const postCommentFailure = error => ({
+  type: POST_COMMENT_FAILURE,
+  payload: {
+    status: 'postcommentFailure',
+    error
+  }
+});
+export const getComments = articleId => async dispatch => {
+  const token = localStorage.getItem('token');
```
Comment (segunolalive) [inline src/store/modules/comments/actions/index.js]: Abstract this functionality to a separate utility.

Checking localstorage is i/o which slow. It's synchronous, meaning it blocks the main thread and prevents you application from doing anything else or responding to user input.

Localstorage operations should also be wrapped in a try/catch block.

**Review by segunolalive — CHANGES_REQUESTED** [review]
Good job 🎆 🕺 

Make the required changes.

File: src/components/Comments/CreateComments.js (hunk 1 above)
Comment (Onyimatics) [reply src/components/Comments/CreateComments.js]: Feedback taken and done!

File: src/components/Comments/CreateComments.js (hunk 2 above)
Comment (Onyimatics) [reply src/components/Comments/CreateComments.js]: Okay

File: src/components/Comments/ViewComments.js (hunk 3 above)
Comment (Onyimatics) [reply src/components/Comments/ViewComments.js]: All right

File: src/store/modules/comments/actions/index.js (hunk 5 above)
Comment (Onyimatics) [reply src/store/modules/comments/actions/index.js]: Done! Thank you

**Review by tohbay — APPROVED** [review]
Great job @Onyimatics 👍 
Looks good to me

**Review by kechyy — APPROVED** [review]
Good job @Onyimatics 

Seen you've implemented the changes given by the TTL which looks good to me.

**Pull request merged**